aatools/food_insecurity/fewsnet.py

```diff
@@ -0,0 +1,218 @@
+"""
+FEWS NET processing.
+
+Download and save the data provided by FEWS NET.
+"""
+
+import logging
+from datetime import datetime
+from pathlib import Path
+
+from aatools.utils.io import download_url, unzip
+
+logger = logging.getLogger(__name__)
+
+
+def download_zip(
+    url: str,
+    zip_path: Path,
+    output_dir: Path,
+) -> bool:
+    """
+    Download and unzip the file at the url.
+
+    Parameters
+    ----------
+    url : str
+        url that contains the zip file to be downloaded
+    zip_path : Path
+        path to which the content of the url should be downloaded
+    output_dir : Path
+        path of dir to which the zip content should be written
+
+    Returns
+    -------
+    valid_file: bool
+        if True, the url contains a valid zip file
+
+    """
+    valid_file = False
+
+    try:
+        download_url(url, zip_path)
+        logger.info(f"Downloaded {url} to {zip_path}")
+
+        try:
+            unzip(zip_path, output_dir)
+            logger.debug(f"Unzipped {zip_path}")
+            valid_file = True
+        except Exception:
+            # indicates that the url returned something that wasn't a
+            # zip, happens often and indicates data for the given
+            # country - date is not available
+            logger.debug(f"No zip data returned from url {url}")
+        # remove the zip file
+        zip_path.unlink()
+
+    except Exception:
+        logger.info(
+            f"Couldn't download url {url}. "
+            f"Check the url and output path: {zip_path}"
+        )
+
+    return valid_file
+
+
+def download_fewsnet_country(
+    date: datetime,
+    iso2: str,
+    output_dir: Path,
+    # question: do you make this an optional arg
+    # if the function is only called by another function
+    # that already has it as optional arg?
+    use_cache: bool,
+) -> bool:
+    """
+    Download fewsnet data that covers the iso2 country.
+
+    Parameters
+    ----------
+    date : datetime
+        date for which the data should be downloaded
+        only the year and month part are used
+        this commonly refers to the month of the Current Situation period
+    iso2 : str
+        iso2 code for which the data should be downloaded
+    output_dir: Path
+        path of dir to which the zip content should be written
+    use_cache: bool
+        if True, don't download if output_dir already exists
+
+    Returns
+    -------
+    country_data : bool
+        if True, country data for the given date and iso2 exists
+    """
+    base_url_country = "https://fdw.fews.net/api/ipcpackage/"
+    url_country = (
+        f"{base_url_country}?country_code={iso2}"
+        f"&collection_date={date.strftime('%Y-%m')}-01"
+    )
+    output_dir_country = output_dir / f"{iso2}{date.strftime('%Y%m')}"
+    # question: is there a better way to define this zip path?
```

Comment:
I'm not sure, do I understand correctly that you're creating a directory with a `.zip` in the directory name?

Reply:
Not entirely. There is the `output_dir_country` which is a directory that the content of the zip file will be written to. And then there is the zip file itself `zip_path_country`, which has the same name as the directory but has the `.zip` extension because it is a zip file. 
But I think I am trying to optimize it too much while loosing clarity. Will change it to use `stem` instead